src/Microsoft.DotNet.Interactive.HttpRequest.Tests/ParserTests.cs

```diff
@@ -0,0 +1,595 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Linq;
+using FluentAssertions;
+using FluentAssertions.Execution;
+using Microsoft.DotNet.Interactive.HttpRequest.Tests.Utility;
+using Xunit;
+
+namespace Microsoft.DotNet.Interactive.HttpRequest.Tests;
+
+public class ParserTests
+{
+
+    public class Trivia
+    {
+        [Fact]
+        public void it_can_parse_an_empty_string()
+        {
+            var result = HttpRequestParser.Parse("");
+            result.SyntaxTree.Should().BeNull();
+
+            // TODO: Test error reporting.
+        }
+
+        [Fact]
+        public void it_can_parse_a_string_with_only_whitespace()
+        {
+            var result = HttpRequestParser.Parse(" \t ");
+
+            result.SyntaxTree.RootNode
+                  .ChildNodes.Should().ContainSingle<HttpRequestNode>().Which
+                  .MethodNode.ChildTokens.First().TextWithTrivia.Should().Be(" \t ");
+
+            // TODO: Test error reporting.
+        }
+
+        [Fact]
+        public void it_can_parse_a_string_with_only_newlines()
+        {
+            var result = HttpRequestParser.Parse("\r\n\n\r\n");
+
+            result.SyntaxTree.RootNode
+                  .ChildNodes.Should().ContainSingle<HttpRequestNode>().Which
+                  .MethodNode.TextWithTrivia.Should().Be("\r\n\n\r\n");
+
+            // TODO: Test error reporting.
+            // Regardless of number of newlines, there should be consistent nodes
+            // set of whitespace, newline, and punctuation
+            //each token kind
+            //combination of characters should be checked by the lexer
+        }
+    }
+
+
+    public class Lexer
+    {
+        [Fact]
+
+        public void multiple_whitespaces_are_treated_as_a_single_token()
+        {
+            var result = HttpRequestParser.Parse("     ");
```

Comment:
Could you please also include a tab character in the middle? The tab character should also be lexed into to the same whitespace token.

Reply:
```suggestion
            var result = HttpRequestParser.Parse("  \t   ");
```